packages/calypso-products/src/get-difm-tiered-price-details.ts

```diff
@@ -0,0 +1,82 @@
+import { isDIFMProduct } from './is-difm-product';
+import type { Product } from './types';
+
+/**
+ * Returns meaningful DIFM purchase details related to tiered difm prices if available
+ * Returns null if this is not a DIFM purchase or the proper related price tier information is not available.
+ *
+ * @param { Product | any } product  product to get details from
+ * @param { number } noOfPages  the number of pages required
+ * @returns {object} with the relevent tier details
+ */
+export function getDIFMTieredPriceDetails(
+	// Any is allowed here because the ProductListItem in (client/state/products-list/selectors/get-products-list.ts) is not compatible with the Product object
+	// And that type cannot be imported into an external module
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	product: Product | any,
```

Comment:
We don't need to use `any` just because the types we need cannot be imported. You can create a new interface in this file which has only the required properties.

For example, it looks like this function only cares about `product_slug` (required) and `price_tier_list` (optional), so this should work:

```ts
export type DIFMPriceTierProduct = Pick< Product, 'product_slug' > &
	Pick< Partial< Product >, 'price_tier_list' >;

export function getDIFMTieredPriceDetails(
	product: undefined | DIFMPriceTierProduct,
	noOfPages: number
)
```

Reply:
Thanks for this suggestion @sirbrillig 💡 
There was an incompatibility occuring from using the product_slug from Product so I went ahead with the below types.

https://github.com/Automattic/wp-calypso/blob/01ab2e6795c908591ee174f2587a6cef0934c79a/packages/calypso-products/src/get-difm-tiered-price-details.ts#L4-L6